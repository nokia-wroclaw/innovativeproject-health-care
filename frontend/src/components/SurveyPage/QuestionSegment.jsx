import React from "react";
import { connect } from "react-redux";
import { Segment, TextArea, SegmentGroup } from "semantic-ui-react";
import { setComment } from "./../../store/actions/currentSurvey";
import Question from "./Question";
import "./checkbox.css";

const QuestionSegment = ({ question, disabled, ...props }) => {
  const isCommentRequired = () => {
    if (question.answer === 0 || question.answer === 1) return true;
    return false;
  };

  return (
    <React.Fragment>
      <SegmentGroup>
        <Segment>
          <Question question={question} />
        </Segment>
        <Segment attached>
          <TextArea
            placeholder={disabled ? "No comment provided" : "Tell us more"}
            className={question.comment ? "" : (question.answer < 2 ? "red-outline" : "")}
            required={isCommentRequired()}
            value={question.comment}
            disabled={disabled}
            onChange={(e, { value }) => {
              props.setComment(question.id, value);
            }}
          />
        </Segment>
      </SegmentGroup>
    </React.Fragment>
  );
};

const mapStateToProps = state => ({
  survey: state.currentSurvey
});

export default connect(
  mapStateToProps,
  { setComment }
)(QuestionSegment);
