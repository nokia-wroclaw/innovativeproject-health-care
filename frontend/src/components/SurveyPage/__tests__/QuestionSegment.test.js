import React from "react";
import { shallow } from "enzyme";
import QuestionSegment from "../QuestionSegment";

describe("QuestionSegment", () => {
  it("should match snapshot", () => {
    expect(shallow(<QuestionSegment question={{}} />)).toMatchSnapshot();
  });
});
