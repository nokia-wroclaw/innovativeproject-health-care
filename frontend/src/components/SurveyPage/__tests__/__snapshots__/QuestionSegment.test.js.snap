// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QuestionSegment should match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <QuestionSegment
    question={Object {}}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": <SegmentGroup
        style={
          Object {
            "marginBottom": 10,
          }
        }
      >
        <Segment>
          <Question
            question={Object {}}
          />
        </Segment>
        <Segment
          attached={true}
        >
          <TextArea
            as="textarea"
            className=""
            placeholder="Tell us more"
            required=""
            rows={3}
          />
        </Segment>
      </SegmentGroup>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <Segment>
            <Question
              question={Object {}}
            />
          </Segment>,
          <Segment
            attached={true}
          >
            <TextArea
              as="textarea"
              className=""
              placeholder="Tell us more"
              required=""
              rows={3}
            />
          </Segment>,
        ],
        "style": Object {
          "marginBottom": 10,
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <Question
              question={Object {}}
            />,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "question": Object {},
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "attached": true,
            "children": <TextArea
              as="textarea"
              className=""
              placeholder="Tell us more"
              required=""
              rows={3}
            />,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "as": "textarea",
              "className": "",
              "placeholder": "Tell us more",
              "required": "",
              "rows": 3,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": [Function],
    },
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <SegmentGroup
          style={
            Object {
              "marginBottom": 10,
            }
          }
        >
          <Segment>
            <Question
              question={Object {}}
            />
          </Segment>
          <Segment
            attached={true}
          >
            <TextArea
              as="textarea"
              className=""
              placeholder="Tell us more"
              required=""
              rows={3}
            />
          </Segment>
        </SegmentGroup>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            <Segment>
              <Question
                question={Object {}}
              />
            </Segment>,
            <Segment
              attached={true}
            >
              <TextArea
                as="textarea"
                className=""
                placeholder="Tell us more"
                required=""
                rows={3}
              />
            </Segment>,
          ],
          "style": Object {
            "marginBottom": 10,
          },
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": <Question
                question={Object {}}
              />,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "question": Object {},
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "attached": true,
              "children": <TextArea
                as="textarea"
                className=""
                placeholder="Tell us more"
                required=""
                rows={3}
              />,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "as": "textarea",
                "className": "",
                "placeholder": "Tell us more",
                "required": "",
                "rows": 3,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
